app/javascript/retrospring/features/moderation/blockAnon.ts

```diff
@@ -0,0 +1,44 @@
+import Rails from '@rails/ujs';
+import swal from 'sweetalert';
+
+import { showErrorNotification, showNotification } from "utilities/notifications";
+import I18n from "retrospring/i18n";
+
+export function blockAnonEventHandler(event: Event): void {
```

Comment:
```suggestion
export function blockAnonEventHandler(event: Event): void {
    event.preventDefault();
    
```

now that the link is a proper `a` tag we need this.